File: validation/src/main/java/com/practice/validation/post/service/PostService.java
package com.practice.validation.post.service;

import com.practice.validation.exception.NotFoundException;
import com.practice.validation.post.domain.Post;
import com.practice.validation.post.repository.PostRepository;
import com.practice.validation.user.domain.User;
import com.practice.validation.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;
    private final UserService userService;
    public Post findByUser(Long userId) {
        User user = userService.findById(userId);
        return postRepository.findByWriter(user).orElseThrow(
                () -> new NotFoundException("user의 게시물이 존재하지 않습니다.")
        );
    }
}

File: sse-kt/src/main/kotlin/org/practice/sse/service/PostService.kt
package org.practice.sse.service

import org.practice.sse.controller.dto.PostCreate
import org.practice.sse.domain.Post
import org.practice.sse.domain.Member
import org.practice.sse.domain.Notification
import org.practice.sse.repository.PostRepository
import org.springframework.stereotype.Service

@Service
class PostService(
    private val userService: MemberService,
    private val notificationService: NotificationService,
    private val postRepository: PostRepository
) {
    fun create(postCreate: PostCreate) {
        val user : Member = userService.findById(1L)
        val post : Post = Post.of(postCreate, user)
        postRepository.save(post)
        //notification 저장
        val notification : Notification = Notification.createPostNotification(user, post.id!!)
        //notification 생성
        notificationService.createNotification(notification)
    }
}